### ISSUE TITLE
[java] SimplifiedTernary: support for the java.lang.Boolean class

### ISSUE THREAD

**Issue by dykov** [pmd/pmd#3604]
**Is your feature request related to a problem? Please describe.**
I faced ternary operators with Boolean objects as result:
```java
condition ? java.lang.Boolean.TRUE : something();
```

**Describe the solution you'd like**
In my opinion, it would be a good solution to disable the use of this class. just like the boolean in the original rule.

**Comment by oowekyala** [comment]
If `something()` returns null and you've refactored to `condition || something()` then your expression NPEs. This is not a problem with a boolean literal (eg `condition ? true : (Boolean) null`) because that code NPEs too.

I think maybe, UnnecessaryBoxing could carefully\* look into whether `Boolean.TRUE` may be unboxed to `true` (which here it can't). If it can then SimplifiedTernary would flag it. I don't think this case should be added to SimplifiedTernary.

\* boxing rules in ternaries with a primitive type are very complicated...
